ui/inspectors/inspector.reel/inspector-confirmation.reel/inspector-confirmation.js

```diff
@@ -8,9 +8,47 @@ var Component = require("montage/ui/component").Component;
  * @extends Component
  */
 exports.InspectorConfirmation = Component.specialize(/** @lends InspectorConfirmation# */ {
-    constructor: {
-        value: function InspectorConfirmation() {
-            this.super();
+
+    handleConfirmDeleteAction: {
+        value: function (event) {
+            this.parentComponent.isConfirmationVisible = false;
+            this.confirmDelete();
+        }
+    },
+
+    handleCancelDeleteAction: {
+        value: function() {
+            this.parentComponent.isConfirmationVisible = false;
+        }
+    },
+
+    confirmDelete: {
+        value: function (event) {
+            var self = this,
+                promise;
+
+            this.parentComponent._isToBeDeleted = true;
+
+            if (typeof this.parentComponent.parentComponent.delete === 'function') {
```

Comment:
Walk up the Component Tree is too fragile. If the structure change in the future, this code won't work.